src/module_3/module_3.md

```diff
@@ -0,0 +1,891 @@
+      <td>ricepastapulses</td>
+      <td>2808429314180</td>
+      <td>3537167515780</td>
+      <td>2020-10-06 10:37:05</td>
+      <td>2020-10-06 00:00:00</td>
+      <td>3</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 27 columns</p>
+</div>
+
+
+
+### Filtering for orders with 5 or more items
+
+
+```python
+items_per_order = df.groupby('order_id')['outcome'].sum()
+big_orders = items_per_order[items_per_order >= 5]
+orders_nums = big_orders.index
+df = df[df['order_id'].isin(orders_nums)]
+```
+
+### Optimize DataFrame for memory
+
+
+```python
+df.info()
+```
+
+    <class 'pandas.core.frame.DataFrame'>
+    Index: 2163953 entries, 0 to 2880547
+    Data columns (total 27 columns):
+     #   Column                            Dtype  
+    ---  ------                            -----  
+     0   variant_id                        int64  
+     1   product_type                      object 
+     2   order_id                          int64  
+     3   user_id                           int64  
+     4   created_at                        object 
+     5   order_date                        object 
+     6   user_order_seq                    int64  
+     7   outcome                           float64
+     8   ordered_before                    float64
+     9   abandoned_before                  float64
+     10  active_snoozed                    float64
+     11  set_as_regular                    float64
+     12  normalised_price                  float64
+     13  discount_pct                      float64
+     14  vendor                            object 
+     15  global_popularity                 float64
+     16  count_adults                      float64
+     17  count_children                    float64
+     18  count_babies                      float64
+     19  count_pets                        float64
+     20  people_ex_baby                    float64
+     21  days_since_purchase_variant_id    float64
+     22  avg_days_to_buy_variant_id        float64
+     23  std_days_to_buy_variant_id        float64
+     24  days_since_purchase_product_type  float64
+     25  avg_days_to_buy_product_type      float64
+     26  std_days_to_buy_product_type      float64
+    dtypes: float64(19), int64(4), object(4)
+    memory usage: 462.3+ MB
+
+
+
+```python
+df.drop(['order_id', 'created_at', 'order_date'], inplace=True, axis=1)
+
+columns = ['user_order_seq', 'outcome', 'ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular', 'count_adults', 'count_children', 'count_babies', 'count_pets', 'people_ex_baby', 'days_since_purchase_variant_id', 'days_since_purchase_product_type']
+for c in columns:
+    df[c] = df[c].astype('int32')
```

Comment:
good idea, but I would make sure that this is not screwing your numbers since can be a little bit dangerous